test/e2e/testsuites/scale_tester.go

```diff
@@ -0,0 +1,359 @@
+	}
+	tStorageClass, storageCleanup := t.Pod.CreateStorageClass(client, namespace, t.CSIDriver, t.StorageClassParameters)
+	defer storageCleanup()
+	for i := 0; i < numberOfTotalSS; i++ {
+		tStatefulSet, cleanupStatefulSet := t.Pod.SetupStatefulset(client, namespace, t.CSIDriver, schedulerName, t.Replicas, t.StorageClassParameters, &tStorageClass)
+		tStatefulSets = append(tStatefulSets, tStatefulSet)
+		for i := range cleanupStatefulSet {
+			i := i
+			defer cleanupStatefulSet[i]()
+		}
+		wg.Add(1)
+		go func(ss *TestStatefulset) {
+			defer wg.Done()
+			defer ginkgo.GinkgoRecover()
+			tokens <- struct{}{} // acquire a token
+			ss.CreateWithoutWaiting()
+			<-tokens // release the token
+		}(tStatefulSet)
+	}
+	wg.Wait()
+
+	numberOfRunningPods := 0
+
+	for {
+		numberOfRunningPods = CountAllPodsWithMatchingLabel(client, namespace, map[string]string{"app": "azuredisk-volume-tester"})
+		if numberOfRunningPods >= totalNumberOfPods {
+			break
+		}
+		timePassed := time.Since(start)
+		switch {
+		case timePassed <= 1*time.Minute:
+			oneMinPods = numberOfRunningPods
+		case timePassed <= 2*time.Minute:
+			twoMinPods = numberOfRunningPods
+		case timePassed <= 3*time.Minute:
+			threeMinPods = numberOfRunningPods
+		case timePassed <= 5*time.Minute:
+			fiveMinPods = numberOfRunningPods
+		case timePassed <= 10*time.Minute:
+			tenMinPods = numberOfRunningPods
+		}
+		time.Sleep(30 * time.Second)
+	}
+	klog.Infof("Test completed in %f minutes.", time.Since(start).Minutes())
+	klog.Infof("1 min: %d pods are ready", oneMinPods)
+	klog.Infof("2 min: %d pods are ready", twoMinPods)
+	klog.Infof("3 min: %d pods are ready", threeMinPods)
+	klog.Infof("5 min: %d pods are ready", fiveMinPods)
+	klog.Infof("10 min: %d pods are ready", tenMinPods)
+}
+
+func (t *PodSchedulingOnFailoverScaleTest) Run(client clientset.Interface, namespace *v1.Namespace, schedulerName string) {
+	nodeNames := ListNodeNames(client)
+	if len(nodeNames) < 1 {
+		ginkgo.Skip("Insufficient node count. Current node count is %d", len(nodeNames))
+	}
+
+	ginkgo.By("Initiating statefulset setup")
+
+	statefulSets := []*TestStatefulset{}
+	for i := range t.Pods {
+		pod := t.Pods[i]
+		tStorageClass, storageCleanup := pod.CreateStorageClass(client, namespace, t.CSIDriver, t.StorageClassParameters)
+		defer storageCleanup()
+
+		statefulSet, cleanup := pod.SetupStatefulset(client, namespace, t.CSIDriver, schedulerName, 1, t.StorageClassParameters, &tStorageClass)
+		for j := range cleanup {
+			defer cleanup[j]()
+		}
+
+		statefulSets = append(statefulSets, statefulSet)
+	}
+
+	ginkgo.By("Completed statefulset setup")
+
+	ginkgo.By("Initiating statefulset deployment")
+	var wg sync.WaitGroup
+	start := time.Now()
+	for i := range statefulSets {
+		wg.Add(1)
+		go func(statefulset *TestStatefulset) {
+			defer wg.Done()
+			defer ginkgo.GinkgoRecover()
+
+			statefulset.Create()
+			statefulset.WaitForPodReady()
```

Comment:
True for all the tests below: `statefulset.Create()` already calls `statefulset.WaitForPodReady()` no need to call it separately. Besides, this is not an efficient way to wait for the scale test completion as it issues too many requests and will cause some more throttling.

Reply:
I had checked the behavior for TestPod and it didnt wait for the creation to succeed. Thanks for pointing it out. Any particular reason for different behaviors between TestPod and TestStatefulSet?

Reply:
Not that I know of. This part was not implemented by our team.